solidity/test/rewards/TestECDSARewards.js

```diff
@@ -0,0 +1,232 @@
+      await verifyIntervalAllocations(5)
+    })
+
+    it("should equal to expected allocations when 1 keep was created per interval", async () => {
+      await verifyIntervalAllocations(1)
+    })
+  })
+
+  async function verifyIntervalAllocations(keepToCreatePerInterval) {
+    let keepCreationTimestamp = firstKeepCreationTimestamp
+
+    for (let i = 0; i < 24; i++) {
+      await createKeeps(keepToCreatePerInterval, keepCreationTimestamp)
+
+      keepCreationTimestamp = await timeJumpToEndOfInterval(i)
+
+      await rewardsContract.allocateRewards(i)
+
+      const actualBalance = (await rewardsContract.getAllocatedRewards(i)).div(
+        tokenDecimalMultiplier
+      )
+
+      expect(actualBalance).to.gte.BN(expectedKEEPAllocation - precision)
+      expect(actualBalance).to.lte.BN(expectedKEEPAllocation + precision)
+    }
+  }
+
+  describe("rewards distribution", async () => {
+    it("should not be possible when a keep is not closed", async () => {
+      await createKeeps(1, firstKeepCreationTimestamp)
+
+      const keepAddress = await keepFactory.getKeepAtIndex(0)
+
+      const isEligible = await rewardsContract.eligibleForReward(keepAddress)
+      expect(isEligible).to.be.false
+
+      await timeJumpToEndOfInterval(0)
+      await expectRevert(
+        rewardsContract.receiveReward(keepAddress),
+        "Keep is not closed"
+      )
+    })
+
+    it("should not be possible when a keep is terminated", async () => {
+      await createKeeps(1, firstKeepCreationTimestamp)
+
+      const keepAddress = await keepFactory.getKeepAtIndex(0)
+      const keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsTerminated()
+
+      const isEligible = await rewardsContract.eligibleForReward(keepAddress)
+      expect(isEligible).to.be.false
+
+      await timeJumpToEndOfInterval(0)
+      await expectRevert(
+        rewardsContract.receiveReward(keepAddress),
+        "Keep is not closed"
+      )
+    })
+
+    it("should not count terminated groups when distributing rewards", async () => {
+      await createKeeps(2, firstKeepCreationTimestamp)
+      // reward for the first interval: 7128000 KEEP
+      // keeps created: 2, but the min is 4 => 7128000 / 4 = 1782000 KEEP per keep
+      // member receives: 1782000 / 3 = 594000 (3 signers per keep)
+      const expectedBeneficiaryBalance = new BN(594000)
+
+      await timeJumpToEndOfInterval(0)
+
+      const keepTerminatedAddress = await keepFactory.getKeepAtIndex(0)
+      const keepTerminated = await BondedECDSAKeepStub.at(keepTerminatedAddress)
+      await keepTerminated.publicMarkAsTerminated()
+
+      const keepAddress = await keepFactory.getKeepAtIndex(1)
+      const keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsClosed()
+
+      await expectRevert(
+        rewardsContract.receiveReward(keepTerminatedAddress),
+        "Keep is not closed"
+      )
```

Comment:
This is checked in the previous test, do we need this assertion here?

Reply:
Not really, was just verifying steps along the way.